[SPARK-15715][SQL] Fix alter partition with storage information in Hive

andrewor14 opened the pull request:
## What changes were proposed in this pull request?

This command didn't work for Hive tables. Now it does:

```
ALTER TABLE boxes PARTITION (width=3)
    SET SERDE 'com.sparkbricks.serde.ColumnarSerDe'
    WITH SERDEPROPERTIES ('compress'='true')
```
## How was this patch tested?

`HiveExternalCatalogSuite`

Commit:
Fix alter partition storage

SparkQA commented:
**[Test build #59777 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59777/consoleFull)** for PR 13453 at commit [`ab7df87`](https://github.com/apache/spark/commit/ab7df87226b69cd97fd804dbf1085bfcbbf25c2e).

Commit:
Add end-to-end test for DDLSuite

SparkQA commented:
**[Test build #59781 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59781/consoleFull)** for PR 13453 at commit [`823479d`](https://github.com/apache/spark/commit/823479d0c8b6d34cba080a1723bd14a13d6e500f).

SparkQA commented:
**[Test build #59777 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59777/consoleFull)** for PR 13453 at commit [`ab7df87`](https://github.com/apache/spark/commit/ab7df87226b69cd97fd804dbf1085bfcbbf25c2e).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/59777/
Test FAILed.

SparkQA commented:
**[Test build #59781 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59781/consoleFull)** for PR 13453 at commit [`823479d`](https://github.com/apache/spark/commit/823479d0c8b6d34cba080a1723bd14a13d6e500f).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/59781/
Test FAILed.

cloud-fan commented:
LGTM

Commit:
Fix tests

Avoid using createMetaPartitionObject, which is not available
in Hive 0.12. This commit basically inlines the contents of
createMetaPartitionObject in Spark.

andrewor14 commented on sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala:
```diff
@@ -779,18 +780,29 @@ private[hive] class HiveClientImpl(
     hiveTable
   }
 
-  private def toHiveViewTable(view: CatalogTable): HiveTable = {
```
not used

SparkQA commented:
**[Test build #59871 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59871/consoleFull)** for PR 13453 at commit [`6f69899`](https://github.com/apache/spark/commit/6f698992a001206631ad6009f1109206cfab3360).

cloud-fan commented:
LGTM, pending jenkins

SparkQA commented:
**[Test build #59871 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59871/consoleFull)** for PR 13453 at commit [`6f69899`](https://github.com/apache/spark/commit/6f698992a001206631ad6009f1109206cfab3360).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/59871/
Test PASSed.

andrewor14 commented:
Merging into master 2.0 thanks.
